packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalBlockNavigationTree as BlockNavigationTree,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { Button } from '@wordpress/components';
+import {
+	useFocusOnMount,
+	useFocusReturn,
+	useInstanceId,
+	useMergeRefs,
+} from '@wordpress/compose';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { closeSmall } from '@wordpress/icons';
+import { ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import { store as editPostStore } from '../../store';
+
+export default function ListViewSidebar() {
+	const { clientIdsTree, selectedBlockClientIds } = useSelect( ( select ) => {
+		const {
+			__unstableGetClientIdsTree,
+			getSelectedBlockClientIds,
+		} = select( blockEditorStore );
+		return {
+			clientIdsTree: __unstableGetClientIdsTree(),
+			selectedBlockClientIds: getSelectedBlockClientIds(),
+		};
+	}, [] );
+	const { setIsListViewOpened } = useDispatch( editPostStore );
+
+	const { clearSelectedBlock, selectBlock } = useDispatch( blockEditorStore );
+	async function selectEditorBlock( clientId ) {
+		await clearSelectedBlock();
```

Comment:
The clear is only relevant if you "re-select" the selected block.

Basically, the behaviour is that selecting a block in the sidebar moves the focus to the canvas.
This happens by observing the state for changes on the selected block.
If we don't clear the selection, selecting a selected block does not update the state, which in turns doesn't trigger the focus shift.

I guess we might only want to clear selection in case of "re-selection", but I think the action is cheap enough to not cause any issues. 🤔